utils/awsFunctions.ts

```diff
@@ -0,0 +1,173 @@
+    }
+  };
+  
+  /**
+   * Creates the user in the Identity store.
+   * @param {string} username - The username to create.
+   * @param {string} email - The email to associate with the user.
+   * @returns {Promise<any>} - The AWS response or error.
+   */
+  export const createUser = async (username: string, email: string): Promise<any> => {
+    const client = configureAWSCredentials();
+  
+    const params: CreateUserCommandInput = {
+      IdentityStoreId: identityStoreId,
+      UserName: username,
+      Name: {
+        Formatted: username,
+        FamilyName: username,
+        GivenName: username,
+      },
+      DisplayName: username,
+      Emails: [
+        {
+          Value: email,
+          Type: "work",
+          Primary: true,
+        },
+      ],
+    };
+  
+    try {
+      const command = new CreateUserCommand(params);
+      const response = await client.send(command);
+      return response;
+    } catch (error) {
+      console.error(`The error from create user ${error}`);
+      throw new Error(`Failed to create user: ${error instanceof Error ? error.message : String(error)}`);
+    }
+  };
+  
+  /**
+   * This function adds the user to the IAM group.
+   * @param {string} groupId - The group ID to add the user to.
+   * @param {string} userId - The user ID to add to the group.
+   * @returns {Promise<any>} - The AWS response or error.
+   */
+export const addUserToGroup = async (groupId: string, awsUserId: string): Promise<any> => {
+    const client = configureAWSCredentials();
+  
+    const params: CreateGroupMembershipCommandInput = {
+      IdentityStoreId: identityStoreId,
+      GroupId: groupId,
+      MemberId: {
+        UserId: awsUserId,
+      },
+    };
+  
+    try {
+      const command = new CreateGroupMembershipCommand(params);
+      const response = await client.send(command);
+      return response;
+    } catch (error) {
+      console.error("Error adding user to group:", error);
+      if (error.__type === 'ConflictException'){
+        return { conflict: true };
+      }
+
+      throw new Error(`Failed to add user to group: ${error instanceof Error ? error.message : String(error)}`);
+    }
+  };
+  
+  /**
+   * Function to get UserId by username from AWS Identity Store.
+   * @param {string} username - The username of the user.
+   * @returns {Promise<string | null>} - The UserId if found, otherwise null.
+   */
+  export const fetchAwsUserIdByUsername = async (username: string): Promise<string | null> => {
+    const client = configureAWSCredentials();
+  
+    const params: ListUsersCommandInput = {
+      IdentityStoreId: identityStoreId,
+      Filters: [
+        {
+          AttributePath: "UserName",
+          AttributeValue: username,
+        },
+      ],
+    };
+  
+    try {
+      const response = await client.send(new ListUsersCommand(params));
+  
+      if (response.Users && response.Users.length > 0) {
```

Comment:
Can there be more than one AWS users associated with a username?

Reply:
No, have used username as in RDS discord we have some script which changes the user name to be unique like `vikhyat-1` and others will get `vikhyat-2` and so on...